src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventIdValidator.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import io.vavr.control.Validation;
+
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+
+import static io.vavr.control.Validation.invalid;
+import static io.vavr.control.Validation.valid;
+import static java.lang.String.format;
+
+public final class EventIdValidator {
+
+    private static final String EVENT_ID_ATTACH_TO_CASE = "attachToExistingCase";
+    private static final String EVENT_ID_CREATE_CASE = "createCase";
+
+    private EventIdValidator() {
+        // utility class constructor
+    }
+
+    @Nonnull
+    static Validation<String, Void> isAttachToCaseEvent(String eventId) {
+        return hasValidEventId(EVENT_ID_ATTACH_TO_CASE::equalsIgnoreCase, eventId);
+    }
+
+    @Nonnull
+    static Validation<String, Void> isCreateCaseEvent(String eventId) {
+        return hasValidEventId(EVENT_ID_CREATE_CASE::equalsIgnoreCase, eventId);
+    }
+
+    @Nonnull
+    private static Validation<String, Void> hasValidEventId(Function<String, Boolean> isValid, String eventId) {
```

Comment:
'valid' or 'expected'?
```
isCreateCaseEvent(EVENT_ID_ATTACH_TO_CASE)`
```
will say `EVENT_ID_ATTACH_TO_CASE` is not supported. But it very much is.

When will this check be used in general?
Perhaps include usage in the PR

Reply:
This is private method. And in source code. So no questions about `isCreateCase(AttachCaseEvent)` integrity possible here nor `expected` - keyword is used in test cases